ServerCore/PuzzleStateHelper.cs

```diff
@@ -185,7 +185,7 @@ public static async Task SetUnlockStateAsync(PuzzleServerContext context, Event
             }
 
             // Award hint coins
-            if (value != null && puzzle.HintCoinsForSolve != 0)
+            if (value != null && puzzle != null && puzzle.HintCoinsForSolve != 0)
```

Comment:
Why would puzzle be null?

Reply:
null means "all puzzles". In theory I would amass all hints in this case but what would you spend them on?